src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -530,14 +530,36 @@ std::shared_ptr<ov::Model> ov::XmlDeserializer::parse_function(const pugi::xml_n
     }
 
     // Run DFS starting from outputs to get nodes topological order
-    std::function<void(size_t)> dfs = [&edges, &order, &dfs_used_nodes, &dfs](const size_t id) {
-        if (dfs_used_nodes.count(id))
-            return;
-        dfs_used_nodes.insert(id);
-        for (auto& edge : edges[id]) {
-            dfs(edge.fromLayerId);
+    auto dfs = [&](size_t start_id) {
+        std::vector<size_t> stack;
+        std::unordered_set<size_t> visited;
```

Comment:
Is it required maybe using stack could be enough to process only what is on stack?